File: src/test/java/uk/co/endofhome/javoice/invoice/InvoiceTest.java
package uk.co.endofhome.javoice.invoice;

import org.junit.Test;
import uk.co.endofhome.javoice.customer.Customer;

import java.time.LocalDate;

import static com.googlecode.totallylazy.Sequences.sequence;
import static com.googlecode.totallylazy.matchers.Matchers.is;
import static org.junit.Assert.assertThat;

public class InvoiceTest {

    @Test
    public void can_get_nett_value() {
        Customer customer = new Customer("", "", "", "", "", "");
        ItemLine firstItemLine = new ItemLine(1.0, "things", 1.0);
        ItemLine secondItemLine = new ItemLine(1.0, "stuff", 1.0);
        Invoice invoice = new Invoice("some num", LocalDate.now(), customer, "", sequence(firstItemLine, secondItemLine));

        assertThat(invoice.nettValue(), is(2.0));
    }

    @Test
    public void can_get_nett_value_when_no_itemLines() {
        Customer customer = new Customer("", "", "", "", "", "");
        Invoice invoice = new Invoice("some num", LocalDate.now(), customer, "", sequence());

        assertThat(invoice.nettValue(), is(0.0));
    }

    @Test
    public void can_get_customer_name() {
        Customer customer = new Customer("Andrew Cyrille", "", "", "", "", "");
        Invoice invoice = new Invoice("", LocalDate.now(), customer, "", sequence());

        assertThat(invoice.customerName(), is("Andrew Cyrille"));
    }
}

File: src/test/java/uk/co/endofhome/javoice/invoice/InvoiceTest.kt
package uk.co.endofhome.javoice.invoice

import com.googlecode.totallylazy.matchers.Matchers.`is`
import org.junit.Assert.assertThat
import org.junit.Test
import uk.co.endofhome.javoice.customer.Customer
import java.time.LocalDate

class InvoiceTest {

    @Test
    fun can_get_nett_value() {
        val customer = Customer("", "", "", "", "", "")
        val firstItemLine = ItemLine(1.0, "things", 1.0)
        val secondItemLine = ItemLine(1.0, "stuff", 1.0)
        val invoice = Invoice("some num", LocalDate.now(), customer, "", listOf(firstItemLine, secondItemLine))

        assertThat(invoice.nettValue(), `is`(2.0))
    }

    @Test
    fun can_get_nett_value_when_no_itemLines() {
        val customer = Customer("", "", "", "", "", "")
        val invoice = Invoice("some num", LocalDate.now(), customer, "", emptyList())

        assertThat(invoice.nettValue(), `is`(0.0))
    }

    @Test
    fun can_get_customer_name() {
        val customer = Customer("Andrew Cyrille", "", "", "", "", "")
        val invoice = Invoice("", LocalDate.now(), customer, "", emptyList())

        assertThat(invoice.customerName(), `is`("Andrew Cyrille"))
    }
}